src/v/redpanda/admin_server.cc

```diff
@@ -3339,3 +3346,51 @@ void admin_server::register_shadow_indexing_routes() {
           return sync_local_state_handler(std::move(req));
       });
 }
+
+ss::future<> admin_server::restart_redpanda_service(service_kind service) {
+    if (service == service_kind::schema_registry) {
+        // Checks specific to schema registry
+        if (_schema_registry == nullptr) {
+            throw ss::httpd::server_error_exception(
+              "Schema Registry is undefined. Is it set in the .yaml config "
+              "file?");
+        }
+
+        try {
+            co_await _schema_registry->restart();
+        } catch (...) {
+            throw ss::httpd::server_error_exception(
+              "Unknown issue restarting schema_registry");
+        }
+    } else {
+        throw ss::httpd::server_error_exception(
+          fmt::format("Restart is unsupported for service: {}", service));
+    }
+}
+
+ss::future<ss::json::json_return_type>
+admin_server::redpanda_services_restart_handler(
+  std::unique_ptr<ss::httpd::request> req) {
+    std::optional<service_kind> service;
+    if (auto node = req->get_query_param("service"); node.size() > 0) {
+        service = from_string_view<service_kind>(node);
+        if (!service.has_value() || *service == service_kind::none) {
+            throw ss::httpd::not_found_exception(
+              fmt::format("Service name is required: {}", *service));
+        }
+    } else {
+        throw ss::httpd::not_found_exception("Service name is required");
+    }
+
+    vlog(logger.info, "Restart a redpanda service");
+    co_await restart_redpanda_service(*service);
+    co_return ss::json::json_return_type(ss::json::json_void());
```

Comment:
I'd be tempted to shuffle the code a bit here, assuming `service_kind::none` is removed, and an empty param is already checked since it's a required param:
```suggestion
    auto param = req->get_query_param("service");
    auto service = from_string_view<service_kind>(param);
    if (*service.has_value()) {
        throw ss::httpd::not_found_exception( 
          fmt::format("Invalid service: {}", param));
    }

    vlog(logger.info, "Restarting redpanda service {}", to_string_view(*service));
    co_await restart_redpanda_service(*service);
    co_return ss::json::json_return_type(ss::json::json_void());
```